File: 백준/Bronze/11653. 소인수분해/소인수분해.js
const fs = require('fs');
const input = Number(fs.readFileSync('/dev/stdin').toString().trim());

const getPrimeFactors = (n) => {
    const factors = [];
    let divisor = 2;
    
    while (n >= divisor * divisor) {
        while (n % divisor === 0) {
            factors.push(divisor);
            n = n / divisor;
        }
        divisor++;
    }

    if (n > 1) {
        factors.push(n);
    }

    return factors;
}

if (input > 1) {
    getPrimeFactors(input).forEach((ele) => console.log(ele));
}

File: 백준/Bronze/11653. 소인수분해/소인수분해.rs
use std::io;

fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let mut n: u32 = input.trim().parse().unwrap();
    
    if n == 1 {
        return;
    }
    
    for factor in 2..=(n as f64).sqrt() as u32 {
        while n % factor == 0 {
            println!("{}", factor);
            n /= factor;
        }
       
    }
    
    if n > 1 {
        println!("{}", n);
    }
}